content/docs/reference-react.md

```diff
@@ -103,16 +103,14 @@ React.cloneElement(
 )
 ```
 
-Clone and return a new React element using `element` as the starting point. The resulting element will have the original element's props with the new props merged in shallowly. New children will replace existing children. `key` and `ref` from the original element will be preserved.
+Clone and return a new React element using `element` as the starting point. The resulting element will have the original element's props with the new props merged in shallowly. New children, if passed as `...children` or otherwise `props.children`, will replace existing children. `key` and `ref` from the original element will be preserved unless `props` contains `key` or `ref` (which will overwrite the `key` or `ref` in the cloned element, respectively).
```

Comment:
The new wording is informative but it reads a little confusingly to me. How do you feel about just slightly tweaking what was there before?

> Clone and return a new React element using `element` as the starting point. The resulting element will have the original element's props with the new props merged in shallowly. New children will replace existing children. `key` and `ref` from the original element will be preserved unless they are explicitly overridden.

Reply:
Hmmm, how can we be precise without being confusing?  "explicitly overridden" doesn't make it clear that they can be overridden by mixing `key` or `ref` in with the other props passed to `cloneElement`.  To me that's not obvious unless explicitly stated because `key` and `ref` are like pseudoprops.

And as far as children goes, "new children will replace existing children" doesn't clarify that existing children can be preserved -- it seems to imply that if `...children` is not provided, then the cloned element will have no children.

The fact that `children` can be overridden via `props.children` is also not obvious at all.   I'm trying to fully and correctly document all behavior of `cloneElement`, which really isn't that trivial...